ENIGMAsystem/SHELL/Graphics_Systems/General/GSscreen.cpp

```diff
@@ -153,7 +153,10 @@ static inline int draw_tiles()
 
 void clear_view(float x, float y, float w, float h, float angle, bool showcolor)
 {
-  d3d_set_projection_ortho(x, y, w, h, angle);
+  if (enigma::d3dMode && enigma::d3dPerspective)
+    d3d_set_projection_perspective(x, y, w, h, angle);
```

Comment:
@JoshDreamland this is where I (and GM) makes `d3d_set_perspective` actually take effect.

Reply:
I might instead suggest storing the default protection in a matrix, although I suppose the parameters might not multiply in cleanly. This approach is therefore fine, but please consider a `draw_clear_projection` method, or maybe `d3d_reset_projection`, or your choice of permutation.

Reply:
That would be a good idea, but we're not exactly set up to do that cleanly yet until I generalize the transforms. I am not entirely sure I understand the suggested functions though.